src/canvas.rs

```diff
@@ -11,16 +11,37 @@ pub struct Canvas {
 }
 
 impl Canvas {
-    pub fn new(width: usize, height: usize, pixels: &[u8]) -> Result<Self, Box<dyn Error>> {
+    pub fn new(width: usize, height: usize, pixels: &[u8]) -> Self {
         let bytespp = pixels.len() / (width * height);
         let line_length = bytespp * width;
-        Ok(Canvas {
+        Canvas {
             pixels: pixels.to_owned(),
             width,
             height,
             line_length,
             bytespp,
-        })
+        }
+    }
+
+    pub fn from_color(width: usize, height: usize, r: u8, g: u8, b: u8) -> Self {
+        let mut pixels = vec![];
+        for _ in 0..width {
+            for _ in 0..height {
+                pixels.push(b);
+                pixels.push(g);
+                pixels.push(r);
+                pixels.push(255);
+            }
+        }
```

Comment:
I get an error from `canvas.copy_from_canvas(&green)`. This does not make a vec of the right size.